src/soroban.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {bigint | number | string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = (amount / 10 ** decimals).toFixed(decimals);
```

Comment:
Thinking about this some more, I'm a bit worried about precision. @aristidesstaffieri, @jeesunikim what do you two think? Is it possible that we'll lose bits by dividing by such large numbers (e.g. `1 / 1e-10`)? Maybe we should string-manipulate here, too, like we do in the inverse function. I tried it out [with a runkit here](https://runkit.com/61833274873f8c000b42e673/64e3c3b97c07390008e92d45), basically relying on this:

```js
return [formatted.slice(0, -decimals), formatted.slice(-decimals)].join('.');
```

to figure out where to insert the `.`.

Reply:
yeah 100% I think that should be the approach, if you look back at the original one we did start by shifting decimal places which I believe is just string manipulation on big numbers.

Reply:
The thing that's a bit uncertain to me with this approach is what to do if (a) there are already `.`s in the input `amount` and (b) if there are zeroes at the end of the `amount` (I elaborate in the FIXME in the runkit), but maybe you guys have already considered these cases?

Reply:
ah well I don't see why zeros at the end would be different than any number at the end? I think the number of decimals only tells you how many chars to move when placing the `.`. As far already being a decimal, before this we basically assume some sanity and don't check for malformed strings that can't become a formatted value, but I think we could check for existing dots, non numeric chars, and others things to throw in case it can't be formatted.

Reply:
Yeah that makes sense to me. Maybe the ambiguity is clearer with an existing decimal:

```js
formatTokenAmount("1230.000", 5) === 12.3 or 0.0123
```

You see what I mean? But it sounds like you're saying just treat the input as-is, irrespective of no-impact trailing zeroes.

Reply:
well I think actually a string with an existing decimal is not valid input for this function, because that would imply that it is already formatted. We never checked for this before but I think this should throw in that case. 

Otherwise I think for trailing zeros we could drop them but also not sure we want to assume that it will always be desired. I wonder if it's better to have an option that lets you toggle wether or not to trim zeros?

Reply:
Yeah, I think we should throw on decimals. And imo we don't need an option until requested; trimming off trailing zeroes is a simple regex that can (probably should) be done by the caller.

Reply:
sweet, works for me. Thanks for being diligent on that one.

Reply:
thank you both. updated.